### DISCUSSION TITLE
Sql views

### DISCUSSION THREAD

**Discussion by iskaa02** [pocketbase/pocketbase#713] [Polls]
Hey folks.
I'm using Pocketbase with a new project, and I'm loving it so far.
The only thing annoys me that it's not possible to write raw SQL queries unless you use it as a framework, it's not a problem for me i know and love Go, it might be worth the effort for writing complex queries but for a small and simple query is not it, not mentioning that you have to recompile and deploy new changes.

I was thinking about adding the ability write raw SQL queries as views, you'd have a dashboard for view, and when creating a new one you'd have an editor with code completion capabilities and can test and deploy the views, when deployed you'd access them the same way you do with collections and adding API rules

I'm already using something like this but with a cli, i write the code to create a new view, execute it , then access them using `api/views` end point.

**Comment by benallfree** [comment]
This is a cool idea, especially if there can be different security for the views. It would mean we could lock the real tables and just expose limited column views. 

> you have to recompile and deploy new changes

As a side note…author of [PBScript](https://github.com/benallfree/pbscript) here. You can write PocketBase server side handlers in Typescript or Javascript with no recompile, plus version rollback and live reload. It is a simpler way to write server-side queries for those who need dynamic queries.
